src/gui/settings/qgssettingseditorwidgetwrapper.h

```diff
@@ -80,6 +82,19 @@ class GUI_EXPORT QgsSettingsEditorWidgetWrapper : public QObject
      */
     virtual void setWidgetFromVariant( const QVariant &value ) const = 0;
 
+    /**
+     * Enables automatic update, which causes the setting to be updated immediately when the widget
+     * value is changed.
+     *
+     * If a \a dialog is provided, the setting will be updated when the dialog is accepted.
+     * If not, the setting will be updated directly at each widget value change.
+     *
+     * \note This must called after createEditor() or configureEditor().
+     *
+     * \since QGIS 3.40
+     */
+    void configureAutomaticUpdate( QDialog *dialog = nullptr );
```

Comment:
Sorry, I think this would be a bit better in my humble opinion :)

(also, you'll need a sip update, so good reason to do a final name tweak ;) )

```suggestion
    void configureUpdateBehavior( QDialog *dialog = nullptr );
```

Reply:
I prefer configureAutomaticUpdate because it actually enables the auto update (not direct if dialog, direct if no dialog)

Reply:
@3nids , I guess yeah automatic update in that it'll automatically take care of updates (live update vs dialog close update). OK.